### PR TITLE
[Thorchain] Support protobuf serialization messages (Stargate)

### LINKED ISSUE #2089: [Thorchain] Cosmos v0.45 SDK upgrade
Thorchain disables standard Cosmos `cosmos-sdk/MsgSend` and uses its own `thorchain/MsgSend` https://gitlab.com/thorchain/thornode/-/blob/develop/proto/thorchain/v1/x/thorchain/types/msg_send.proto

```protobuf
syntax = "proto3";
package types;

option go_package = "gitlab.com/thorchain/thornode/x/thorchain/types";

import "cosmos/base/v1beta1/coin.proto";
import "gogoproto/gogo.proto";

message MsgSend {
  bytes from_address = 1  [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
  bytes to_address = 2  [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
  repeated cosmos.base.v1beta1.Coin amount = 3 [(gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins", (gogoproto.nullable) = false];
}
```

The difference is: `from_address` and `to_address` type is `bytes` not `string`

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Instead of the legacy Json serialization (which is currently outdated, see #2089), support protobuf serialized Cosmos messages (Stargate).
Note that Thorchain uses its own message format (from and to addresses are bytes-encoded instead of string (contain key hashes instead of string address), and own type).
Fixes #2089 .

Note: Thorchain-specific message is included within Cosmos (along with a Terra-specific message), separating out is not trivial; addressed in #2093 .

## Testing instructions

Thorchain unit tests, Thorchain RUNE transfers.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: src/proto/Cosmos.proto [hunk 1]
```diff
@@ -119,6 +119,14 @@ message Message {
         repeated string coin = 6;
     }
 
+    // thorchain/MsgSend
+    message THORChainSend {
+        bytes from_address = 1;
```
Comment (hewigovens) [inline src/proto/Cosmos.proto:124]: Let's use string like other messages?

File: src/proto/Cosmos.proto [hunk 2]
```diff
@@ -119,6 +119,14 @@ message Message {
...
+        bytes to_address = 2;
+        repeated Amount amounts = 3;
+        string type_prefix = 4;
```
Comment (hewigovens) [inline src/proto/Cosmos.proto:127]: Does it need? I think we can hardcode for thorchain

**Review by hewigovens — APPROVED** [review]

File: src/proto/Cosmos.proto (hunk 1 above)
Comment (catenocrypt) [reply src/proto/Cosmos.proto:124]: In the proto the bytes address contains the key hash, not the Bech32 string version.  Therefore I would keep that here as well; string is not right for binary keyhash.

File: src/proto/Cosmos.proto (hunk 2 above)
Comment (catenocrypt) [reply src/proto/Cosmos.proto]: Ok.  Prefix is not needed for protobuf version, anyhow.

**Pull request merged**